test/test_theme_data_reader.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require "helper"
+
+class TestThemeDataReader < JekyllUnitTest
+
+  context "site without a theme" do
+    setup do
+      @site = fixture_site("theme" => nil)
+      @reader = Reader.new(@site)
+      @reader.read_data
```

Comment:
`@reader` isn't used subsequently.
Therefore, call the reader directly:
```suggestion
      @site.reader.read_data
```